### PR TITLE
Fix for "get_app_store_version_number" for apps with limited availability

### PR DESCRIPTION
**Pull request by stef-t**
I've been using `get_app_store_version_number` with the `bundle_id` parameter and came across the following error, although the provided bundle id matched the bundle id of my published app on the AppStore: 
 > Cannot find app with #{bundle_id} bundle ID in the App Store

Trying with different bundle ids, I realized that an app's availability influences the result of the itunes lookup API:
- App available in multiple countries: itunes lookup API finds the app.
- App available in one country: itunes lookup API cannot find the app. (The app in question is currently only available to my local country.)

When supplying the url parameter _country_ with a country code to the itunes lookup API call, it will find both apps!

The pull requests, adds a new parameter `country` to `get_app_store_version_number`. If `country` is passed as parameter, it is added as url parameter value when calling the itunes lookup API.

### CODE REVIEW

File: README.md
Comment (jdouglas-nz) [inline README.md:123]: I think it might be a little clearer if we also gave an invocation example without country code - just as to hopefully not mislead people. Thoughts?

File: lib/fastlane/plugin/versioning/actions/get_app_store_version_number.rb
Comment (jdouglas-nz) [inline lib/fastlane/plugin/versioning/actions/get_app_store_version_number.rb:23]: any chance of writing a little test for this? there is some sort of example of these inside of `get_app_store_version_number_spec.rb` - and if you aren't sure how to run the tests - cd into your local `fastlane-plugin-versioning` repo checkout, and run `rake spec`

**Review by jdouglas-nz — COMMENTED** [review]
nice stuff! a cool little change - just a couple hopefully little things to do 👍

File: lib/fastlane/plugin/versioning/actions/get_app_store_version_number.rb
Comment (stef-t) [reply lib/fastlane/plugin/versioning/actions/get_app_store_version_number.rb]: I've added some new commits to the pull request. One of them now contains a test case, which runs `get_app_store_version_number` with the new `country` parameter.

This includes a response I've downloaded from the itunes lookup API and added to the project. The new test case and response mockup handling should be in line with existing test cases.

There is just one problem remaining when running the new test case: The JSON parser throws an error, when parsing the mock response from the itunes lookup API. _I've verified that the JSON is **valid** with multiple validators._
Unfortunately I'm out of ideas on how to proceed. I even updated the JSON gem to the latest version. Could you take a look?

File: README.md
Comment (stef-t) [reply README.md]: That's true. To avoid confusion I've added different invocation examples to the readme. Please let me know what you think.

File: spec/get_app_store_version_number_spec.rb
Comment (jdouglas-nz) [inline spec/get_app_store_version_number_spec.rb:33]: thank you! noice 👍

**Review by jdouglas-nz — APPROVED** [review]
nice one! this looks great to me.

**Comment by stef-t** [comment]
Thank you! Did you have a chance to look at the JSON parser error I mentioned above?

**Comment by jdouglas-nz** [comment]
@stef-t Sorry, I haven't yet -- I'll have a dig through when I get a moment !

**Comment by jdouglas-nz** [comment]
@stef-t - I may of led you awry slightly by saying the existing request is a good example - might be some magic in the headers (maybe content length actually) - but I simply re-`cURL`-ed the request for ya and it worked ! Go figure 🤷‍♂️

**Comment by stef-t** [comment]
Must be some header magic then. Good thing it was so easy to fix for you!

**Comment by jdouglas-nz** [comment]
@stef-t - once you resolve the merge conflict we can get this merged in! 🎉

**Comment by stef-t** [comment]
> @stef-t - once you resolve the merge conflict we can get this merged in! 🎉

Done, thank you!

**Comment by jdouglas-nz** [comment]
@stef-t - I'll merge it in unless you want to and push out a gem the next few hours. Ty again for your contribution!

**Pull request merged**
